hcpl_recursion/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+#[macro_export]
+macro_rules! _recursion__let_rec {
+    ($f:ident = |$($arg_id:ident: $arg_ty:ty),*| -> $ret:ty $body:block) => {
+        let $f = {
+            pub trait AlmostFTrait {
+                fn call(&self, $($arg_id: $arg_ty),*) -> $ret;
+            }
+            pub struct Almost<F>(F);
```

Comment:
`pub` er unødvendig her

```suggestion
            trait AlmostFTrait {
                fn call(&self, $($arg_id: $arg_ty),*) -> $ret;
            }
            struct Almost<F>(F);
```